tests/clients/test_fenda_ditto_client.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import torch
+from flwr.common import Config
+
+from fl4health.clients.fenda_ditto_client import FendaDittoClient
+from fl4health.model_bases.fenda_base import FendaModel
+from fl4health.model_bases.sequential_split_models import SequentiallySplitExchangeBaseModel
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from tests.clients.fixtures import get_client  # noqa
+from tests.test_utils.models_for_test import FeatureCnn, FendaHeadCnn, HeadCnn
+
+
+@pytest.mark.parametrize("type,model", [(FendaDittoClient, FendaModel(FeatureCnn(), FeatureCnn(), FendaHeadCnn()))])
+def test_setting_initial_weights(get_client: FendaDittoClient) -> None:  # noqa
+    torch.manual_seed(42)
+    fenda_ditto_client = get_client
+    fenda_ditto_client.global_model = SequentiallySplitExchangeBaseModel(FeatureCnn(), HeadCnn())
+    fenda_ditto_client.parameter_exchanger = FullParameterExchanger()
+    config: Config = {"current_server_round": 1}
+
+    params = [val.cpu().numpy() + 1.0 for _, val in fenda_ditto_client.global_model.state_dict().items()]
+    fenda_ditto_client.set_parameters(params, config, fitting_round=True)
+    fenda_ditto_client.update_before_train(1)
+
+    global_model_params = [
+        val.detach().clone()
+        for name, val in fenda_ditto_client.global_model.state_dict().items()
+        if name.startswith("base_module.")
```

Comment:
With the initialize_all_weights functionality, we should match the params for all components of the `global_model_params` not just those in the base_module, if I'm not mistaken?

Reply:
yes! removing the `if...`